tests/groups/test_db.py

```diff
@@ -0,0 +1,85 @@
+from virtool.fake.next import DataFaker
+from virtool.groups.transforms import AttachPrimaryGroupTransform, AttachGroupsTransform
+from virtool.data.transforms import apply_transforms
+
+
+async def create_fake_users_with_groups(
```

Comment:
**Points (if you were going to retain this design)**
* This will break if someone ever passes `use_primary_group=True` without passing `faker`. Always require `faker`.
* You could make this function a fixture so you can directly inject `fake2` instead of passing it from the test function. You can define another function in the fixture body that takes your other args, then return the inside function.
* The method `fake2.users.create()` can do a better job of what you are doing here. It has keyword arguments to do what you want.
* You can [avoid flags](https://martinfowler.com/bliki/FlagArgument.html) completely by changing the design.

**To do**

* Get rid of all arguments (eg. `faker`, `quantity`, etc). Create a selection of users that have the characteristics you want across your tests and use the same set of users in each test. This reduces complexity and makes the test more robust because you don't have to carefully construct a different set of data using flags for each test case.

  You can then get the raw user documents you need for each test case using `mongo`.
* Make this a fixture. Rename to a more noun-y name instead of verb-y since it isn't a callable anymore.
* Let `fake2` take care of the user IDs instead of creating your own using `i`.